[knowledge-base] startapp & add data models

yizou275 opened the pull request:
This PR mainly implements the data models according to https://github.com/uchicago-cs/chigame/wiki/Knowledge-Base-~-Tables.md. Details include:
- startapp knowledge_base, update INSTALLED_APPS in src/config/settings/base.py
- update knowledge_base/models.py, including `Guide`, `ReviewFeedback`, `Favorite`, `Like`
- add `published_guide_id` field to `Games` object in games/models.py
- add `moderator` boolfield to `User` object in users/models.py

Closes: #514

Commit:
[knowledge-base] startapp & add data models

Finished the following:
- startapp, update INSTALLED_APPS in src/config/settings/base.py
- update knowledge_base/models.py, based on wiki page
- add published_guide_id field to Games object in games/models.py
- add moderator boolfield to User object in users/models.py

Closes: #514

Commit:
[knowledge-base] fix github pr check

Changes name in knowledge-base/apps.py to fix failing check,
CI / run-tests (pull_request).

Fixes: PR #546.

Luca-W-D reviewed (changes requested):
Really great work! Super detailed – I love the behavior that calling .save() on a Guide object has conditional logic for updating the timestamp. A ton of work clearly went into this PR.

I cloned the branch locally and looked into the test debug log. I think the tests are failing because the knowledge-base DB migrations weren't pushed to the remote repo. After `make-migrations` and `migrate`, this branch works as expected! Happy to look again after that file/those files are pushed.

Commit:
[knowledge-base] fix pr check by adding migrations

Add migrations for modified models to fix failing check,
CI / run-tests (pull_request).

Fixes: PR #546.

yizou275 commented:
Thanks Luca for your insight! I just pushed the migration files. The check seems to pass now:)

Luca-W-D commented:
Great! I have only one mechanical note:
 * Right now, you have likes and favorites implemented as explicit tables. I think that it may be easier for us to implement these as a many-to-many relationship. This creates a table under the hood anyway, and it allows for easier interactions in code. Specifically, this would allow us to call something like `guide.likes.count()` so we don't have to maintain `n_likes` as a separate field. Does this sound right to you?

Also, this is just nit-picky, but the examples we've covered in class have defined integer choices to be a subclass of the model they're used in. I'd recommend moving GuideStatus to be under Guide (so that if we ever call `Guide` in code we can immediately reference `Guide.GuideStatus.PENDING` or similar). Of course, your call on whether to change it!

Commit:
[knowledge-base] addressed pr feedback

- change likes and favorites from explicit tables to
manytomany field
- put GuideStatus inside Guide object

Closes: #514

yizou275 commented:
These are great points! I addressed both in my new commit. Thank you so much:)

Luca-W-D reviewed (approved):
Looks good to me! I was going to comment that the explicit integer choices under ReviewFeedback may be verbose, but I realized that it allows us to avoid a "pending" response to a review. Again, great detail! :)

jacqsmei reviewed (approved):
Great job on this! I agree with the notes Luca left.

yizou275 committed:
Merge branch 'dev' into knowledge-base/implement-django-models

Commit:
[knowledge-base] solves migration conflict

Resolve migration conflict during merging. Should
pass the pr test now.

Fixes: PR #546.

jacqsmei commented:
Issue Score: Excellent

Comments:
Nice job on this one + great feedback/review

<!-- {"score": "excellent", "minor": false} -->
